portal_tables/unify_grant_tables.py

```diff
@@ -0,0 +1,145 @@
+    parser.add_argument("-p2", "--project2",
+                        type=str, required=True,
+                        help=("CCKP - MC2 Admin Synapse ID"))
+    parser.add_argument("--dryrun", action="store_true")
+    return parser.parse_args()
+
+def getManifestsFromGrant(syn,theParent):
+    """This function examines the GrantView table
+    and returns a list of all Synapse table IDs
+    separated by manifest type (publications, datasets,
+    tools) for each grant"""
+
+    # Access each grant project by accessing 'grantId'
+    # column in grantView manifest
+    # synId for grantView is hardcoded in and
+    # will need to be changed for future use
+    df = syn.tableQuery(f"select {'grantId'} from {theParent}").asDataFrame()
+    gIds = df['grantId'].tolist()
+
+    # for each grant, get publications,
+    # datasets, and tools 
+
+    # value inside dictionary will reset upon each
+    # loop iteration, so appending to this list 
+    # maintains each one
+    mTypes = []
+    mIds = []
+    for gId in gIds:
+        grantX = syn.get(gId)
+        # By accessing the grants, it is possible
+        # to access their 'children'
+
+        # extract manifest type and table id from generator
+        mTypes.append([m['name'] for m in syn.getChildren(grantX, includeTypes=['table'])])
+        mIds.append([m['id'] for m in syn.getChildren(grantX, includeTypes=['table'])])
+
+    # with each synapse table id mapped to a type of manifest
+    # it is possible to begin adding them to the staging table
+    # but should flatten list first for easier key-val mapping
+    flat_mTypes = list(itertools.chain(*mTypes))
+    flat_mIds = list(itertools.chain(*mIds))
+
+    idToManType = dict(zip(flat_mIds, flat_mTypes))
+    print(idToManType)
+
+    # use dictionary of synapse id to manifest type
+    # to prepare to separate data into rows
+    pubsId = []
+    datasetsId = []
+    toolsId = []
+
+    for key,val in idToManType.items():
+        if val == 'publications':
+            pubsId.append(key)
+        elif val == 'datasets':
+            datasetsId.append(key)
+        else:
+            toolsId.append(key)
+    return pubsId, datasetsId, toolsId
+
+def getEachManifestFromCCKP(syn, theNewParent):
+    """this function reads in the Synapse ID of the
+    CCKP Admin table and finds the table IDs of the
+    PublicationView, DatasetView, and ToolView manifests"""
+
+    pMan = syn.findEntityId('PublicationView', theNewParent)
+    dMan = syn.findEntityId('DatasetView', theNewParent)
+    tMan = syn.findEntityId('ToolView', theNewParent)
+
+    return pMan, dMan, tMan
+
+def main():
+    syn = synapseclient.Synapse()
+    syn.login()
+    
+    args = get_args() 
+    originTable = args.project1
+    destinationTable = args.project2
+
+    # these are the lists of each kind of manifest pulled from
+    # all individual grants
+    [allPub, allData, allTool] = getManifestsFromGrant(syn, originTable)
```

Comment:
I don't think the brackets here are necessary, therefore, can be removed for readability.